src/sentry/integrations/api/endpoints/organization_coding_agents.py

```diff
@@ -11,20 +17,109 @@
 from sentry.api.bases.organization import OrganizationEndpoint
 from sentry.constants import ObjectStatus
+from sentry.integrations.coding_agent.integration import CodingAgentIntegration
+from sentry.integrations.coding_agent.models import CodingAgentLaunchRequest
 from sentry.integrations.coding_agent.utils import get_coding_agent_providers
 from sentry.integrations.services.integration import integration_service
+from sentry.models.organization import Organization
+from sentry.net.http import connection_from_url
+from sentry.seer.autofix.utils import (
+    AutofixState,
+    AutofixTriggerSource,
+    CodingAgentState,
+    get_autofix_state,
+    get_coding_agent_prompt,
+)
+from sentry.seer.signed_seer_api import make_signed_seer_api_request
 
 logger = logging.getLogger(__name__)
 
 
+# Follows the GitHub branch name rules:
+# https://docs.github.com/en/get-started/using-git/dealing-with-special-characters-in-branch-and-tag-names#naming-branches-and-tags
+# As our coding agent integration only supports launching on GitHub right now.
+VALID_BRANCH_NAME_CHARS = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_/"
+
+
+def sanitize_branch_name(branch_name: str) -> str:
+    """
+    Sanitize a branch name by taking the first 3 words, converting to lowercase,
+    removing/replacing special characters, and adding 6 unique random characters.
+
+    Args:
+        branch_name: The raw branch name to sanitize
+
+    Returns:
+        A sanitized branch name safe for use as a Git branch
+    """
+    # Take only the first 3 words
+    words = branch_name.strip().split()[:3]
+    truncated_name = " ".join(words)
+
+    # Although underscores are allowed, we standardize to kebab case.
+    kebab_case = truncated_name.replace(" ", "-").replace("_", "-").lower()
+    sanitized = "".join(c for c in kebab_case if c in VALID_BRANCH_NAME_CHARS)
+    sanitized = sanitized.rstrip("/")
+
+    # Generate 6 unique random characters (alphanumeric)
+    # This is to avoid potential branch name conflicts.
+    random_suffix = "".join(
+        secrets.choice(string.ascii_lowercase + string.digits) for _ in range(6)
+    )
+
+    # Combine sanitized name with random suffix
+    return f"{sanitized}-{random_suffix}" if sanitized else f"branch-{random_suffix}"
+
+
+def store_coding_agent_state_to_seer(run_id: int, coding_agent_state: CodingAgentState) -> None:
+    """Store coding agent state via Seer API."""
+    path = "/v1/automation/autofix/coding-agent/state"
+    body = orjson.dumps(
+        {
+            "run_id": run_id,
+            "coding_agent_state": coding_agent_state.dict(),
+        }
+    )
+
+    connection_pool = connection_from_url(settings.SEER_AUTOFIX_URL)
+    response = make_signed_seer_api_request(
+        connection_pool,
+        path,
+        body=body,
+        timeout=30,
+    )
+
+    if response.status >= 400:
+        raise Exception(f"Seer API error: {response.status}")
```

Comment:
same Q as previous PR: can we raise a more specific exception (you can inherit from `Exception` to do so, or inherit from  `SentryAPIException` if you want to format something for http clients / the browser to expect